src/app/components/create-new-device/create-new-device.component.ts

```diff
@@ -0,0 +1,91 @@
+import { ConfirmationDialogComponent } from './../confirmation-dialog/confirmation-dialog.component';
+import { DeviceService } from './../../services/device.service';
+import { Component, OnInit } from '@angular/core';
+import { Subject } from 'rxjs';
+import { FormGroup, FormControl, Validators, FormBuilder } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { MatDialog } from '@angular/material';
+import { MatSnackBar } from '@angular/material';
+
+@Component({
+  selector: 'app-create-new-device',
+  templateUrl: './create-new-device.component.html',
+  styleUrls: ['./create-new-device.component.css']
+})
+export class CreateNewDeviceComponent implements OnInit {
+  ngUnsubscribe = new Subject();
+  createDeviceForm: FormGroup;
+  submitted: boolean;
+  static PATH: any = 'devices/create';
+
+  constructor(private createDeviceService: DeviceService, private formBuilder: FormBuilder,
+              private route: ActivatedRoute, private router: Router, public dialog: MatDialog,
+              private snackBar: MatSnackBar) { }
+
+  ngOnInit() {
+    this.route.queryParams.subscribe(params => {
```

Comment:
Move this to the constructor so it happens before the html loads.

Reply:
All fixed with the newest commit.